### PR TITLE
Introducing new version API 2019-05-01-preview with adding a new filter for usageDetails called publisherType

### PR DESCRIPTION
**Pull request by parisa-naeimi**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/consumption/resource-manager/readme.md [hunk 1]
```diff
@@ -26,18 +26,27 @@ These are the global settings for the Consumption API.
...
+```yaml $(tag) == 'package-preview-2019-05'
+input-file:
+  - Microsoft.Consumption/preview/2019-05-01-preview/consumption.json
+```
 ### Tag: package-preview-2019-04
 
 These settings apply only when `--tag=package-preview-2019-04` is specified on the command line.
 
-```yaml $(tag) == 'package-preview-2019-04'
+``` yaml $(tag) == 'package-preview-2019-04'
```
Comment (erich-wang) [inline specification/consumption/resource-manager/readme.md]: Is there any particular reason to add space?

File: specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
```
Comment (erich-wang) [inline specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json]: Please format this json file

**Review by erich-wang — CHANGES_REQUESTED** [review]

**Comment by erich-wang** [comment]
Please also address CI error:

> parameter 'ln' is defined in global parameters section without 'x-ms-parameter-location' extension. This would add the parameter as the client property. Please ensure that this is exactly you want. If so, apply the extension "x-ms-parameter-location": "client". Else, apply the extension "x-ms-parameter-location": "method". at specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json:3177

File: specification/consumption/resource-manager/readme.md (hunk 1 above)
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/readme.md]: I did not do any manual editing here. I just got the latest version and used openapi to generate a new version 2019-05-01-preview. But I can remove it anyways

**Comment by parisa-naeimi** [comment]
> Please also address CI error:
> 
> > parameter 'ln' is defined in global parameters section without 'x-ms-parameter-location' extension. This would add the parameter as the client property. Please ensure that this is exactly you want. If so, apply the extension "x-ms-parameter-location": "client". Else, apply the extension "x-ms-parameter-location": "method". at specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json:3177

File: specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json (hunk 2 above)
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json]: Done, please let me know if it looks good now.

File: specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json [hunk 3]
```diff
@@ -0,0 +1,3186 @@
...
+            "$ref": "#/parameters/scopeUsageDetailsParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/metricParameter"
+          },
+		  {
+            "$ref": "#/parameters/languageParameter"
```
Comment (ryansbenson) [inline specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json]: Why are you introducing an explicit language parameter to this API? All ARM requests are supposed to respect the Accept-Language header as the requested language from the user.

https://github.com/Azure/azure-resource-manager-rpc/blob/70d10c715342149a3be94ebdc75ebde5ea0c9354/v1.0/common-api-details.md#client-request-headers

Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json]: In usageDetailsDownload the API caller needs to pass" ln=en" for instance as a url parameter. Then it is required. We are giving the URL to the CSV file we need to prepare the file based on the caller language specific. In previous PR it was not there, then I added it.

Comment (ryansbenson) [reply specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json]: I understand this is how your API was documented. To be compliant with ARM RPC, you should just use the Accept-Language header from the caller. Having an explicit 'ln' query parameter isn't needed. If you didn't make this change, whats the expected behavior when these 2 languages aren't the same?

Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json]: @ryansbenson: Thank you for the review and I removed language parameter from usage detail download and related examples based on your suggestion to be compatible with ARM.

**Comment by ms-premp** [comment]
@ravbhatnagar can you please review this.
This is same as the older version we had. We are continuing with preview-version as discussed last time and discussed changes will be pulled in a new GA version. That work is still pending and hence this change is still in sync with the current preview version.

File: specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json (hunk 3 above)
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-05-01-preview/consumption.json]: @ryansbenson could you please review this again, I addressed your comments.

**Review by KrisBash — APPROVED** [review]
LGTM, thanks

**Comment by parisa-naeimi** [comment]
@erich-wang could you please let me know when can you check this, I already addressed your comments and ARM already signed off.

**Comment by parisa-naeimi** [comment]
> LGTM, thanks

**Review by erich-wang — APPROVED** [review]

**Pull request merged**